src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ImmutableArrayExtensions.cs

```diff
@@ -2,13 +2,204 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Collections.Internal;
 
 namespace Roslyn.Utilities
 {
     internal static partial class ImmutableArrayExtensions
     {
+        public static int FindIndex<T>(this ImmutableArray<T> items, Predicate<T> match)
```

Comment:
this seems like a ton of helpers to be adding.

Reply:
I was really surprised this didn't already exist. It's the standard overloads for `FindIndex` and `FindLastIndex`.

Reply:
➡️ Switched to different approach